### ISSUE TITLE
Inequality comparison with string literal shows editor error

### ISSUE THREAD

**Issue by yiochen** [Microsoft/TypeScript#14983]
I think this is a bug. I might be wrong.
**TypeScript Version:**  2.2.1

**Code**

```ts
class Greeter {
    greeting: string;
    constructor(message: string) {
        if (message !== 'world' || message !== 'me') {
            // do something
        }
        this.greeting = message;
    }
    greet() {
        return "Hello, " + this.greeting;
    }
}
```

**Expected behavior:**
I am trying this on [typescript playground](https://www.typescriptlang.org/play/), it should show no error in the editor.
**Actual behavior:**
The second condition in the if statement is underlined. The error message is 
```
Operator '!==' cannot be applied to types '"world"' and '"me"'.
```
Same thing happened in Visual studio code with Typescript 2.2.1

**Comment by RyanCavanaugh** [comment]
The condition you have here
```ts
if (message !== 'world' || message !== 'me') {
```
always evaluates to `true`.

Perhaps you meant to write
```ts
if (message !== 'world' && message !== 'me') {
                        ~~
```

**Comment by yiochen** [comment]
Oh, it's because of that. I wish the error message was more intuitive. But thanks, I'm going to close this issue.
